lua/roslyn/init.lua

```diff
@@ -271,14 +211,22 @@ function M.setup(config)
             local predicted_sln_file = require("roslyn.slnutils").predict_sln_file(opt.buf)
 
             if predicted_sln_file then
-                run_roslyn(cmd, predicted_sln_file, roslyn_config.config, roslyn_config.filewatching)
+                wrap_roslyn(predicted_sln_file, roslyn_config)
                 known_solutions[sln_directory] = predicted_sln_file
             end
 
             vim.notify_once(
                 "Multiple sln files found. You can use `CSTarget` to select target for buffer",
                 vim.log.levels.INFO
             )
+
+            vim.api.nvim_buf_create_user_command(opt.buf, "CSTarget", function()
+                vim.ui.select(all_sln_files, { prompt = "Select target solution: " }, function(sln_file)
+                    known_solutions[sln_directory] = sln_file
+                    wrap_roslyn(sln_file, roslyn_config)
+                    vim.api.nvim_buf_del_user_command(opt.buf, "CSTarget")
```

Comment:
I am a bit sceptical to delete the command after using it. What if I select the wrong target for the buffer and then I want to change it again?

Reply:
The change to make it scoped to the buffer makes sense though!

Reply:
yeah, this makes sense, deleting is not the best idea. Fixed in https://github.com/seblj/roslyn.nvim/pull/25/commits/d4cdda1d9ba2fd5c60c2662e94f9d062bc6c1bf2